services/src/main/java/org/apache/druid/guice/CoordinatorServiceModule.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.guice;
+
+import com.google.inject.multibindings.ProvidesIntoSet;
+import com.google.inject.name.Named;
+import org.apache.druid.discovery.DruidService;
+import org.apache.druid.discovery.NoopService;
+
+public class CoordinatorServiceModule extends AbstractDruidServiceModule
+{
+  public static final String COORDINATOR_SERVICE_KEY = "coordinatorService";
+
+  @ProvidesIntoSet
+  @Named(COORDINATOR_SERVICE_KEY)
+  public Class<? extends DruidService> getDefaultNodeService()
+  {
+    return NoopService.class;
+  }
```

Comment:
I was not sure how to bind an empty set in a map binder. So, this default dummy service was for always providing something so that the set is not empty. But it turns it's easy to bind an empty set in a map binder without this dummy service. I have removed it now.